CONTRIBUTING.md

```diff
@@ -0,0 +1,92 @@
+# Contributing
+
+When contributing to this repository, please first discuss the change you wish to make via issue,
+email, or any other method with the owners of this repository before making a change. 
+
+Please note we have a code of conduct, please follow it in all your interactions with the project.
+
+## Pull Request Process
+
+1. Ensure any install or build dependencies are removed before the end of the layer when doing a 
+   build.
+2. Update the README.md with details of changes to the interface, this includes new environment 
+   variables, exposed ports, useful file locations and container parameters.
+3. Increase the version numbers in any examples files and the README.md to the new version that this
+   Pull Request would represent. The versioning scheme we use is [SemVer](http://semver.org/).
+4. You may merge the Pull Request in once you have the sign-off of one other developer, or if you 
+   do not have permission to do that, you may request the reviewer to merge it for you.
+
```

Comment:
We should push passing builds and PGP signing - unless squashing will take care of the necessity to have each commit signed. 

In terms of commits - we should not accept commits that are illicit, inappropriate, or lacking detail. Commits should serve to provide a good idea of the work going on vs. "Fix" or an emoji.

We should also detail the fact that builds must be passing for a pull request to be merged in - and that if you have added new functionality, you should provide tests for it - if not already available.